application/src/ext/java/org/opentripplanner/ext/stopconsolidation/DecorateConsolidatedStopNames.java

```diff
@@ -51,6 +55,46 @@ private void replaceConsolidatedStops(Itinerary i) {
     });
   }
 
+  /**
+   * Removes walk legs from and to a consolidated stop if they are deemed "short". This means that
+   * they are from a different element of the consolidated stop.
+   */
+  private void removeShortWalkLegs(Itinerary itinerary) {
+    var legs = new ArrayList<>(itinerary.getLegs());
+    var first = legs.getFirst();
+    if (
+      service.isPartOfConsolidatedStop(first.getTo().stop) &&
+        isShortWalkLeg(first)
+    ) {
+      legs.removeFirst();
+    }
+    var last = legs.getLast();
+    if (
+      service.isPartOfConsolidatedStop(last.getFrom().stop) &&
+        isShortWalkLeg(last)
+    ) {
+      legs.removeLast();
+    }
+
+    var transfersRemoved = legs.stream().filter(l -> !isTransferWithinConsolidatedStop(l)).toList();
+
+    itinerary.setLegs(transfersRemoved);
+  }
+
+  private static boolean isShortWalkLeg(Leg leg) {
+    return leg.isWalkingLeg() &&
+      leg.getDistanceMeters() < MAX_INTRA_STOP_WALK_DISTANCE_METERS;
+  }
+
+  private boolean isTransferWithinConsolidatedStop(Leg l) {
+    return (
+      l.isWalkingLeg() &&
+      (l.getDistanceMeters() < MAX_INTRA_STOP_WALK_DISTANCE_METERS) &&
```

Comment:
Reuse `isShortWalkLeg` here.

Reply:
Done in https://github.com/opentripplanner/OpenTripPlanner/pull/6157/commits/09ae95d01fd5a796602e506d08fbd7de58b196cf